delivery/media/watermarking/lib/media-delivery-watermarking.js

```diff
@@ -0,0 +1,2134 @@
+        if (null == contentLength || 0 === contentLength.length) throw new Error("Side car processing failed due to no content-length response header found!");
+        const paceinfoLength = parseInt(contentLength[0]), dataArr = await async function(stream, size) {
+            const buffer = new ArrayBuffer(size), arr = new Uint8Array(buffer);
+            let currentPos = 0;
+            return await stream.pipeTo(new WritableStream({
+                write(chunk) {
+                    arr.set(chunk, currentPos), currentPos += chunk.length;
+                }
+            })), arr;
+        }(paceInfoResponse.body, paceinfoLength);
+        return CborParser.decode(dataArr);
+    }
+    getSubVariantPath(variantSubPaths, tmidVariant) {
+        for (let i = 0; i < variantSubPaths.length; i++) {
+            const variantSubPath = variantSubPaths[i];
+            if (null !== variantSubPath && variantSubPath.variant == tmidVariant) return variantSubPath.subPath;
+        }
+        return null;
+    }
+    validateClaims(wmPayload) {
+        if (this.wmOptions.issuer && wmPayload.iss && this.wmOptions.issuer !== wmPayload.iss) throw new Error(`${this.wmOptions.tokenType} malformed: invalid iss, expected ${this.wmOptions.issuer}`);
+        if (this.wmOptions.subject && wmPayload.sub && this.wmOptions.subject !== wmPayload.sub) throw new Error(`${this.wmOptions.tokenType} malformed: invalid sub, expected ${this.wmOptions.subject}`);
+        if (this.wmOptions.audience) {
+            const aud = wmPayload.aud;
+            if (aud && (Array.isArray(aud) || "string" == typeof aud)) {
+                if (!(Array.isArray(aud) ? aud : [ aud ]).includes(this.wmOptions.audience)) throw new Error(`${this.wmOptions.tokenType} malformed: invalid aud, expected ${this.wmOptions.audience}`);
+            }
+        }
+        const clockTimestamp = Math.floor(Date.now() / 1e3);
+        if (!1 === this.wmOptions.ignoreExpiration) {
+            if (!wmPayload.exp) throw new Error(`${this.wmOptions.tokenType} malformed: missing exp field in payload`);
+            if ("number" != typeof wmPayload.exp) throw new Error(`${this.wmOptions.tokenType} malformed: exp must be number`);
+            if (clockTimestamp > wmPayload.exp + (this.wmOptions.clockTolerance || 0)) throw new Error(`${this.wmOptions.tokenType} expired`);
+        }
+        if (!1 === this.wmOptions.ignoreNotBefore) {
+            if (!wmPayload.nbf) throw new Error(`${this.wmOptions.tokenType} malformed: missing nbf field in payload`);
+            if ("number" != typeof wmPayload.nbf) throw new Error(`${this.wmOptions.tokenType} malformed: nbf must be number`);
+            if (wmPayload.nbf > clockTimestamp + (this.wmOptions.clockTolerance || 0)) throw new Error(`${this.wmOptions.tokenType} not active`);
+        }
+    }
+    async importCryptoKeys(keys, authToken) {
```

Comment:
I suspect that caching the crypto keys, rather than re-importing them on every request would be more efficient.

Reply:
I feel if re importing the keys have more performance overhead than the gain we obtain by our cache service,  removing caching would allow us to accept cryptoKeys directly which solves all the problem.

Reply:
> Also customer could have a logic to determine which subset of keys to be used for which token, this wont be possible if we cache it as the module wont know which key to be used for verification.
e.g rs256token1 = keys1[], hs256token2 = keys2[]

Isn't that information encoded in the token? 

If not, the user could create multiple objects.